File: 1259/1259.cpp
/* ************************************************************************** */
/*                                                                            */
/*                                                        :::      ::::::::   */
/*   1259.cpp                                           :+:      :+:    :+:   */
/*                                                    +:+ +:+         +:+     */
/*   By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+        */
/*                                                +#+#+#+#+#+   +#+           */
/*   Created: 2024/02/17 16:49:55 by jihoolee          #+#    #+#             */
/*   Updated: 2024/02/17 17:11:54 by jihoolee         ###   ########.fr       */
/*                                                                            */
/* ************************************************************************** */

#include <iostream>
#include <string>

bool is_palindrome(int num) {
  std::string num_str = std::to_string(num);

  for (size_t idx = 0; idx < num_str.length() / 2; ++idx)
    if (num_str[idx] != num_str[num_str.length() - 1 - idx])
      return false;
  return true;
}

int main(void) {
  int num;

  while (true) {
    std::cin >> num;
    if (num == 0)
      break;
    if (is_palindrome(num))
      std::cout << "yes" << std::endl;
    else
      std::cout << "no" << std::endl;
  }
  return 0;
}

File: 1259/1259.py
# **************************************************************************** #
#                                                                              #
#                                                         :::      ::::::::    #
#    1259.py                                            :+:      :+:    :+:    #
#                                                     +:+ +:+         +:+      #
#    By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+         #
#                                                 +#+#+#+#+#+   +#+            #
#    Created: 2024/02/17 17:12:34 by jihoolee          #+#    #+#              #
#    Updated: 2024/02/17 17:18:30 by jihoolee         ###   ########.fr        #
#                                                                              #
# **************************************************************************** #

def is_palindrome(num_str: str):
    for idx in range(len(num_str) // 2):
        if num_str[idx] != num_str[len(num_str) - 1 - idx]:
            return False
    return True


def main() -> None:
    while (True):
        num: str = input()
        if num == '0':
            break
        if is_palindrome(num):
            print('yes')
        else:
            print('no')


if __name__ == '__main__':
    main()
